src/webui/api/rsscontroller.cpp

```diff
@@ -139,3 +166,26 @@ void RSSController::rulesAction()
 
     setResult(jsonObj);
 }
+
+void RSSController::matchingArticlesAction()
+{
+    requireParams({"ruleName"});
+
+    const QString ruleName {params()["ruleName"].trimmed()};
+    const QList<RSS::Article *> articles {RSS::AutoDownloader::instance()->matchingArticles(ruleName)};
+
+    QJsonObject jsonObj;
+    for (const auto &article : articles) {
+        QString feedName = article->feed()->name();
+        if (!jsonObj.contains(feedName)) {
+            QJsonArray newArray {article->title()};
+            jsonObj.insert(feedName, newArray);
+        }
+        else {
+            QJsonArray keyArray = jsonObj[feedName].toArray();
+            keyArray.push_back(article->title());
+            jsonObj.insert(feedName, keyArray);
```

Comment:
Seems counterproductive to make copy of array at each insertion.
Anyway you can avoid this by eliminating of redundant `AutoDownloader::matchingArticles()` method.